compiler/src/iree/compiler/GlobalOptimization/ConvertStridedContractionToContraction.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/GlobalOptimization/Passes.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/IR/AffineExpr.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir::iree_compiler::GlobalOptimization {
+
+#define GEN_PASS_DEF_CONVERTSTRIDEDCONTRACTIONTOCONTRACTIONPASS
+#include "iree/compiler/GlobalOptimization/Passes.h.inc"
+
+namespace {
+
+class ConvertStridedContractionToContraction
+    : public OpRewritePattern<linalg::GenericOp> {
+public:
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(linalg::GenericOp op,
+                                PatternRewriter &rewriter) const override {
+    // Check if the generic op satisfies all other conditions for being a
+    // contraction.
+    if (op.getNumDpsInputs() != 2 || op.getNumDpsInits() != 1)
+      return failure();
+    if (op.getNumReductionLoops() == 0)
+      return failure();
+    if (!mlir::linalg::detail::isContractionBody(
+            *op.getBlock(), [](Operation *first, Operation *second) {
+              if ((isa<arith::MulFOp>(first) && isa<arith::AddFOp>(second)) ||
+                  (isa<arith::MulIOp>(first) && isa<arith::AddIOp>(second)))
+                return true;
+              return false;
+            }))
+      return failure();
+
+    SmallVector<AffineMap> mapRange = op.getIndexingMapsArray();
+    unsigned inputPos = op.getDpsInputOperand(0)->getOperandNumber();
+    unsigned filterPos = op.getDpsInputOperand(1)->getOperandNumber();
+    unsigned resInitPos = op.getDpsInitOperand(0)->getOperandNumber();
+    AffineMap inputMap = mapRange[inputPos];
+    AffineMap filterMap = mapRange[filterPos];
+    AffineMap resultMap = mapRange[resInitPos];
+    // For now, we are only handling the case where the first input is the
+    // only non-projected permutation.
+    if (!filterMap.isProjectedPermutation() ||
+        !resultMap.isProjectedPermutation())
+      return failure();
+    if (inputMap.isProjectedPermutation())
+      return failure();
+    auto staticShape = op.getStaticLoopRanges();
+
+    llvm::SmallDenseMap<unsigned, int64_t> strides;
+    SmallVector<AffineExpr> replacementExprs;
+    Value input = op.getDpsInputs()[0];
+    auto inputTy = dyn_cast<RankedTensorType>(input.getType());
+    if (!inputTy)
+      return failure();
+    SmallVector<int64_t> inputShape(inputTy.getShape());
+    replacementExprs.reserve(inputMap.getNumResults());
+    // walk through input map and look for expressions of the form `dim * cst`
```

Comment:
Capitalize and add punctuation here and below.
```suggestion
    // Walk through input map and look for expressions of the form `dim * cst`.
```